pkg/liqo-controller-manager/external-network/utils/getters.go

```diff
@@ -124,3 +126,25 @@ func KindToResource(kind string) string {
 	// lowercased and pluralized
 	return strings.ToLower(kind) + "s"
 }
+
+// ResourceToKind returns the kind name for a given resource.
+func ResourceToKind(gvr schema.GroupVersionResource, config *rest.Config) (string, error) {
+	dClient, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return "", err
+	}
```

Comment:
Can you use the client inside the factory? It contains f.KubeClient.Discovery().ServerGroupsAndResources().